src/daft-io/src/lib.rs

```diff
@@ -187,13 +190,22 @@ type Result<T, E = Error> = std::result::Result<T, E>;
 pub struct IOClient {
     source_type_to_store: tokio::sync::RwLock<HashMap<SourceType, Arc<dyn ObjectSource>>>,
     config: Arc<IOConfig>,
+    unity_catalog: Option<Arc<UnityCatalog>>,
```

Comment:
From a glance it seems like we could put the `UnityCatalog` inside `IOConfig` and avoid many of the changes to `io_client`. Is there a reason it needs to be on the client instead of in the config?

Reply:
also, on that note. Is it possible to just store the configuration to create the python `UnityCatalog`? instead of storing an already created instance? I think that'd fit in much better with our existing `IOConfig`.

Reply:
Good idea. I'll change it to that